android/core/src/main/java/com/novoda/peepz/ImageFreshnessPeepComparator.java

```diff
@@ -0,0 +1,36 @@
+package com.novoda.peepz;
+
+import java.util.Comparator;
+
+class ImageFreshnessPeepComparator implements Comparator<Peep> {
+
+    @Override
+    public int compare(Peep peep1, Peep peep2) {
+        if (peep1.image() != null && peep2.image() != null) {
+            return compare(peep1.image(), peep2.image());
+        }
+
+        if (peep1.image() != null && peep2.image() == null) {
+            return -1;
+        }
+
+        if (peep1.image() == null && peep2.image() != null) {
+            return 1;
+        }
+
+        return 0;
```

Comment:
Mmm this can never happen isn't it?
You could simplify to

```
if (peep1.image() != null && peep2.image() != null) {
    ...
} else if (peep1.image() != null) {
    ...
} else {
    ...
}
```